src/unit_tests/monitord/test_monitor_actions.c

```diff
@@ -0,0 +1,384 @@
+    mond_time_control.thismonth = 0;
+    mond_time_control.thisyear = 0;
+
+    return 0;
+}
+
+int teardown_monitord(void **state) {
+    test_mode = 0;
+
+    mond.global.agents_disconnection_alert_time = 0;
+    mond.global.agents_disconnection_time = 0;
+
+    mond.delete_old_agents = 0;
+    mond.a_queue = -1;
+
+    mond_time_control.disconnect_counter = 0;
+    mond_time_control.alert_counter = 0;
+    mond_time_control.delete_counter = 0;
+    mond_time_control.today = 0;
+    mond_time_control.thismonth = 0;
+    mond_time_control.thisyear = 0;
+
+    return 0;
+}
+
+// Tests
+
+/* Tests monitor_send_deletion_msg */
+
+void test_monitor_send_deletion_msg_success(void **state) {
+    char *agent = "Agent1-any";
+    char msg_to_send[OS_SIZE_1024];
+
+    snprintf(msg_to_send, OS_SIZE_1024, OS_AG_REMOVED, agent);
+    mond.a_queue = 1;
+
+    expect_string(__wrap_SendMSG, message, msg_to_send);
+    expect_string(__wrap_SendMSG, locmsg, ARGV0);
+    expect_value(__wrap_SendMSG, loc, LOCALFILE_MQ);
+    will_return(__wrap_SendMSG, 1);
+
+    monitor_send_deletion_msg(agent);
+
+    assert_int_equal(1, mond.a_queue);
+}
+
+void test_monitor_send_deletion_msg_fail(void **state) {
+    char *agent = "Agent1-any";
+    char msg_to_send[OS_SIZE_1024];
+
+    snprintf(msg_to_send, OS_SIZE_1024, OS_AG_REMOVED, agent);
+    mond.a_queue = 1;
+
+    expect_string(__wrap_SendMSG, message, msg_to_send);
+    expect_string(__wrap_SendMSG, locmsg, ARGV0);
+    expect_value(__wrap_SendMSG, loc, LOCALFILE_MQ);
+    will_return(__wrap_SendMSG, -1);
+
+    expect_string(__wrap__mdebug1, formatted_msg, "Could not generate removed agent alert for 'Agent1-any'");
+    expect_string(__wrap__merror, formatted_msg, QUEUE_SEND);
+
+    monitor_send_deletion_msg(agent);
+
+    assert_int_equal(-1, mond.a_queue);
+}
+
+/* Tests monitor_send_disconnection_msg */
+
+void test_monitor_send_disconnection_msg_success(void **state) {
+    char *agent = "Agent1-any";
+    char msg_to_send[OS_SIZE_1024];
+    char header[OS_SIZE_256];
+
+    expect_string(__wrap_wdb_find_agent, name, "Agent1");
+    expect_string(__wrap_wdb_find_agent, ip, "any");
+    will_return(__wrap_wdb_find_agent, 1);
+
+    snprintf(msg_to_send, OS_SIZE_1024, AG_DISCON_MSG, agent);
+    snprintf(header, OS_SIZE_256, "[%03d] (%s) %s", 1, "Agent1", "any");
+
+    expect_string(__wrap_SendMSG, message, msg_to_send);
+    expect_string(__wrap_SendMSG, locmsg, header);
+    expect_value(__wrap_SendMSG, loc, SECURE_MQ);
+    will_return(__wrap_SendMSG, 1);
+    mond.a_queue = 1;
+
+    monitor_send_disconnection_msg(agent);
+
+    assert_int_equal(1, mond.a_queue);
+}
+
+void test_monitor_send_disconnection_msg_agent_removed(void **state) {
+    char *agent = "Agent1-any";
+    char msg_to_send[OS_SIZE_1024];
+
+    expect_string(__wrap_wdb_find_agent, name, "Agent1");
+    expect_string(__wrap_wdb_find_agent, ip, "any");
+    will_return(__wrap_wdb_find_agent, -2);
+
+    // monitor_send_deletion_msg()
```

Comment:
This commented code should be removed.

Reply:
Now the comment is **monitor_send_deletion_msg** without _()_ to make it clear that the test entered in another function